### PR TITLE
More flexible build process for Athena

### PR DESCRIPTION
**Pull request by undernagruzez**
### Motivation and Context
Athena was initially designed as a modular application with a common module (Athena) that is integral to every other module. With multiple students working asynchronously on different modules, stability issues can arise due to database changes, version upgrades, or feature enhancements. This PR leverages Poetry to enable dynamic importing of the common module, Athena, which eliminates the need for a monolithic structure and enhances modularity.

### Description
1. Each module imports the common Athena module dynamically from either a GitHub repository or a local version.
2. Modules are assembled independently using the provided lint all, lock all, and build all scripts.
3. Developers can specify the required version of the middleware for their module.
4. This PR lays the groundwork for pytest integration.

### Steps for Testing
Deploy the changes to a test server.
Verify that all modules function correctly and that the integration with the common Athena module is stable.

### Resolves
This PR resolves the following issues: https://github.com/ls1intum/Athena/issues/321, https://github.com/ls1intum/Athena/issues/298, https://github.com/ls1intum/Athena/issues/297

### CODE REVIEW

**Comment by FelixTJDietrich** [comment]
Enea currently updated the env files for the text llm module on the test server so that module might not work with this PR right now because of this

File: install_modules.py
Comment (FelixTJDietrich) [inline install_modules.py:1]: Maybe we can put all those in a scripts folder? I.e. install_modules, lint_modules, lock_modules, test_modules

File: install_modules.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        "assessment_module_manager",
+        "athena",
+        "log_viewer",
+        "module_example",
+        "module_programming_llm",
+        "module_text_llm",
+        "module_text_cofee",
+        "module_programming_themisml",
+        "module_programming_apted",
+        "module_modeling_llm",
```
Comment (FelixTJDietrich) [inline install_modules.py]: Max mentioned that you might want to also move the modules in subfolders, in that case you could automatically find them? 

I guess we could have a folder structure like:
```
modules/
  text/
    cofee/
    llm/
  programming/
    themisml/
    llm/
    apted/
  modelling/
    llm/
```

We can also do this in a follow-up

Comment (undernagruzez) [reply install_modules.py]: I will do this in this PR

File: install_modules.py
Comment (undernagruzez) [reply install_modules.py:1]: Yes, sure

Comment (undernagruzez) [reply install_modules.py:1]: done

File: install_modules.py (hunk 1 above)
Comment (undernagruzez) [reply install_modules.py]: done

**Comment by FelixTJDietrich** [comment]
Can you also resolve the conflicts?

**Comment by FelixTJDietrich** [comment]
`module_example` is also a programming module

File: .idea/runConfigurations/module_programming_llm.xml
Comment (maximiliansoelch) [inline .idea/runConfigurations/module_programming_llm.xml:1]: I don't think this change should be committed as it includes personal paths

File: .prospector.yaml
Comment (maximiliansoelch) [inline .prospector.yaml:1]: Double check why Athena-coffee is not changed

File: modules/text/module_text_cofee/module_text_cofee/.gitignore
Comment (maximiliansoelch) [inline modules/text/module_text_cofee/module_text_cofee/.gitignore:1]: Double Check

**Review by maximiliansoelch — DISMISSED** [review]

**Comment by FelixTJDietrich** [comment]
@undernagruzez ping me on Slack once this is ready, we should merge this ASAP since it causes a lot of merge conflicts

File: modules/modeling/module_modeling_llm/pyproject.toml [hunk 2]
```diff
@@ -16,12 +17,14 @@ gitpython = "^3.1.41"
 replicate = "^0.11.0"
 tiktoken = "^0.4.0"
 langsmith = "^0.0.63"
+pydantic = "1.10.17"
+prospector = "^1.10.2"
```
Comment (FelixTJDietrich) [inline modules/modeling/module_modeling_llm/pyproject.toml:23]: Why is this no longer a dev dependency?

Comment (FelixTJDietrich) [reply modules/modeling/module_modeling_llm/pyproject.toml:23]: Applies to all modules

**Review by FelixTJDietrich — CHANGES_REQUESTED** [review]
Update `launch.json` and `athena-workspace.code-workspace`

**Review by FelixTJDietrich — APPROVED** [review]
LGTM

**Comment by FelixTJDietrich** [comment]
Also works on TS1 except for programming llm, idk what's up with that.

**Pull request merged**